### ISSUE TITLE
bug: [clickhouse] SQL generating bug

### ISSUE THREAD

**Issue by rongxinyu** [ibis-project/ibis#6838]
### What happened?

Wrong SQL query is generated. Below is how to produce

1. create two tables in clickhouse

`'create table prod.table1 (a int, b int) engine=MergeTree order by a'`
`'create table prod.table2 (c int, d int) engine=MergeTree order by c'`

2. generate SQL with ibis

`con = clickhouse.connect()`
`t1 = con.table('table1')`
`t2 = con.table('table2')`
`t3 = t1.join(t2, t1.a == t2.c)`
`q = t3.mutate(e = t3.c / (t3.a-t3.b))`
`ibis.to_sql(q)`

Below is the generated SQL
```
SELECT
  t2.*,
  t2.c / (
    t2.b - t2.a
  ) AS e
FROM prod.table1 AS t0
INNER JOIN prod.table2 AS t1
  ON t0.a = t1.c
```

Running the above SQL with `q.execute()` has the following error
`Unknown qualified identifier: t2: While processing t2.*, t2.c / (t2.b - t2.a) AS e. (UNKNOWN_IDENTIFIER)`

### What version of ibis are you using?

6.0.0

### What backend(s) are you using, if any?

clickhouse

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by rongxinyu** [comment]
It seems that there is a shortcut for the issue by modifying the join as follows

```
con = clickhouse.connect()
t1 = con.table('table1')
t2 = con.table('table2')
t3 = t1.join(t2.relabel({'c': 'a'}), 'a')
q = t3.mutate(t3.d / (t3.a-t3.b))
ibis.to_sql(q)
```

Below is the new SQL generated, which works
```
SELECT
  *,
  t4.d / (
    t4.b - t4.a
  ) AS "Divide(d, Subtract(b, a))"
FROM (
  SELECT
    t0.a,
    t0.b,
    t2.d
  FROM prod.table1 AS t0
  INNER JOIN (
    SELECT
      t1.c AS a,
      t1.d
    FROM prod.table2 AS t1
  ) AS t2
    ON t0.a = t2.a
) AS t4
```

**Comment by cpcloud** [comment]
@rongxinyu Thanks for the issue!

I am looking into it.

**Comment by cpcloud** [comment]
This will be fixed by #7580.

**Comment by cpcloud** [comment]
Fixed in `the-epic-split`
